core/modules/parser/QSMySqlListener.cc

```diff
@@ -3198,13 +3269,48 @@ class LogicalOperatorAdapter :
     }
 
     void onExit() override {
-        if (_ctx->AND() != nullptr) {
+        if (_ctx->AND() != nullptr || _ctx->getText() == "&&") {
```

Comment:
Even though, I understand what this is about, a difference between `AND` and `&&` still looks a bit confusing. Any chance the later be automatically converted into an object returned by the former?

Reply:
There is not a way to make "and", "AND", and "&&" return a lexical `AND` token. There are ways to  adjust the grammar so that these 3 strings could have their own grammar element which would then get its own Adapter subclass. The pattern would then be duplicated for `OR`. This seems like overkill. The logical operator types are defined as:
```
logicalOperator
    : AND | '&' '&' | XOR | OR | '|' '|'
```
This encapsulates the logic of the types of `AND` in this one location (and also for the types of `OR`), which seems good enough encapsulation for me. (Note, the `AND` token here is case-insensitive). If the difference between the text "and" and "&&" starts spreading then some more abstraction would be correct, but I'm pretty sure it won't happen.

Reply:
Got it. Thanks!